### DISCUSSION TITLE
KLU in NLP for solving KKT unsymmetric systems

### DISCUSSION THREAD

**Discussion by DimaLPorNLP** [DrTimothyAldenDavis/SuiteSparse#885] [Ideas]
Suppose I would like to use KLU to solve the original unsymmetric KKT system (before the elimination of duals to enforce 
symmetry) at every iteration of a NLP problem. Are there cases where KLU is expected to outperform LDLT factorizations included in MUMPS or MA?? (??:=27/57/97)  solvers? What happens with stability, scalings, inertia?

**Comment by DrTimothyAldenDavis** [comment]
It's impossible to say in general since this kind of relative performance depends on the characteristics of the sparse matrix at hand, and they can vary widely, even in the same application.

However, KLU is designed for the very-sparse case, when the matrix is very very sparse and stays that way during factorization.  MUMPS and the MA*7 solvers are designed for the cases when the fillin in the matrix is more substantiall.
